src/appsettings/settings.py

```diff
@@ -1153,3 +1160,62 @@ def check(self):
     def transform(self, value):
         """Transform each item in the list."""
         return tuple(self.inner_setting.transform(item) for item in value)
+
+
+class FileSetting(Setting):
+    """
+    File setting.
+
+    Value of this setting is a pathlib.Path instance.
+    """
+
+    def __init__(
+        self,
+        name="",
+        default=None,
+        required=False,
+        prefix="",
+        call_default=True,
+        transform_default=False,
+        validators=(),
+        exists=True,
+        read_perm=True,
+        write_perm=False,
+    ):
+        """
+        Initialization method.
+
+        Note that attributes ``exists`` and ``mode`` are used in ``FileValidator``. For more information check its
+        documentation.
+
+        Args:
+            name (str): the name of the setting.
+            default (object): default value given to the setting.
+            required (bool): whether the setting is required or not.
+            prefix (str):
+                the setting's prefix (overrides ``AppSettings.Meta`` prefix).
+            call_default (bool): whether to call the default (if callable).
+            transform_default (bool): whether to transform the default value.
+            validators (list of callables): list of additional validators to use.
+            exists (bool): Check whether or not does the file exist.
+            read_perm (bool): Check whether or not do we have read permissions.
+            write_perm (bool): Check whether or not do we have write permissions.
+        """
+        super().__init__(
+            name=name,
+            default=default,
+            required=required,
+            prefix=prefix,
+            call_default=call_default,
+            transform_default=transform_default,
+            validators=validators,
+        )
+        self.exists = exists
+        self.validators.append(FileValidator(exists, read_perm, write_perm))
+
+    def transform(self, value):
+        """Transform value to Path instance."""
+        value = super().transform(value)
+        if not isinstance(value, Path):
+            value = Path(value)
```

Comment:
It seems the `Path` can wrap itself. so the condition is not necessary.

Reply:
Fixed.